R/forward_selection.R

```diff
@@ -8,8 +8,8 @@
 #'
 #' @param scorer  A custom user-supplied function that accepts X and y
 #'   (as defined below) as input and returns the error of the datasets.
-#' @param X tibble. training dataset
-#' @param y tibble. test dataset
+#' @param X tibble. explanatory variables
+#' @param y tibble. target
```

Comment:
1. I think we should stick with the more generic term "data frame". Our package doesn't require the `tibble` package.
2. I copied this from `simulated_annealing.R` for consistency

```suggestion
#' @param X data frame of training features
#' @param y data frame of training targets
```